From d65c61e5b71437ea19ecf632c51e4e8660d4f37d Mon Sep 17 00:00:00 2001
From: "Hebbar, Gururaja" <gururaja.hebbar@ti.com>
Date: Thu, 3 May 2012 15:22:40 +0530
Subject: [PATCH 0945/1092] ARM: OMAP2+: am335x: correct mmc0 pin-mux for
 Beaglebone

Beaglebone Boards has a Micro-sd slot which doesn't support
write-protect pins. But, while beaglebone was setup in the code, common
mmc0 pin-mux was used which had WP pin-mux.

This patch corrects the same by only setting common mmc0 pin mux + cd
only pin-mux

Signed-off-by: Hebbar, Gururaja <gururaja.hebbar@ti.com>
---
 arch/arm/mach-omap2/board-am335xevm.c | 15 ++++++++++++---
 1 file changed, 12 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-omap2/board-am335xevm.c b/arch/arm/mach-omap2/board-am335xevm.c
index 970fb1b..f673c91 100644
--- a/arch/arm/mach-omap2/board-am335xevm.c
+++ b/arch/arm/mach-omap2/board-am335xevm.c
@@ -1595,9 +1595,18 @@ static void d_can_init(int evm_id, int profile)
 
 static void mmc0_init(int evm_id, int profile)
 {
-	setup_pin_mux(mmc0_common_pin_mux);
-	setup_pin_mux(mmc0_wp_only_pin_mux);
-	setup_pin_mux(mmc0_cd_only_pin_mux);
+	switch (evm_id) {
+	case BEAGLE_BONE_A3:
+	case BEAGLE_BONE_OLD:
+		setup_pin_mux(mmc0_common_pin_mux);
+		setup_pin_mux(mmc0_cd_only_pin_mux);
+		break;
+	default:
+		setup_pin_mux(mmc0_common_pin_mux);
+		setup_pin_mux(mmc0_cd_only_pin_mux);
+		setup_pin_mux(mmc0_wp_only_pin_mux);
+		break;
+	}
 
 	omap2_hsmmc_init(am335x_mmc);
 	return;
-- 
1.7.11.2

